airflow-core/src/airflow/assets/manager.py

```diff
@@ -243,15 +288,91 @@ def _queue_dagruns(cls, asset_id: int, dags_to_queue: set[DagModel], session: Se
         # "fallback" to running this in a nested transaction. This is needed
         # so that the adding of these rows happens in the same transaction
         # where `ti.state` is changed.
-        if not dags_to_queue:
+        if get_dialect_name(session) == "postgresql":
+            return cls._queue_dagruns_nonpartitioned_postgres(asset_id, non_partitioned_dags, session)
+        return cls._queue_dagruns_nonpartitioned_slow_path(asset_id, non_partitioned_dags, session)
+
+    @classmethod
+    def _queue_partitioned_dags(
+        cls,
+        asset_id: int,
+        partition_dags: Iterable[DagModel],
+        event: AssetEvent,
+        partition_key: str | None,
+        session: Session,
+    ) -> None:
+        if partition_dags and not partition_key:
+            # TODO: AIP-76 how to best ensure users can see this? Probably add Log record.
+            log.warning(
+                "Listening dags are partition-aware but run has no partition key",
+                listening_dags=[x.dag_id for x in partition_dags],
+                asset_id=asset_id,
+                run_id=event.source_run_id,
+                dag_id=event.source_dag_id,
+                task_id=event.source_task_id,
+            )
             return
```

Comment:
I feel this logic should exist in line 269. but non-blocking

Reply:
ok i'll make a todo